forte/data/extractor/predictor.py

```diff
@@ -0,0 +1,199 @@
+from forte.data.extractor.extractor import TextExtractor, AnnotationSeqExtractor
+from forte.data.extractor.converter import Converter
+from forte.process_manager import ProcessJobStatus
+from forte.data import slice_batch
+from forte.data.data_utils_io import merge_batches, batch_instances
+
+
+class Predictor(BaseProcessor):
+    def __init__(self):
+        super().__init__()
+        self.pack_pools = []
+        self.instance_pools = []
+        self.features_pools = []
+
+    def initialize(self, resources: Resources, configs: Optional[Config]):
+        super().initialize(resources, configs)
+        # TODO: load these params from outside
+        self.batch_size = 11
+        self.scope = Sentence
+
+        config1 = {
+            "scope": Sentence,
+            "entry": Token,
+            "vocab_use_unk": True
+        }
+        extractor1 = TextExtractor(config1)
+        converter1 = Converter(need_pad=True)
+
+        config2 = {
+            "scope": Sentence,
+            "entry": EntityMention,
+            "attribute": "ner_type",
+            "based_on": Token,
+            "strategy": "BIO",
+            "vocab_use_unk": True
+        }
+        extractor2 = AnnotationSeqExtractor(config2)
+        extractor2.add_entry(("PER", "B"))
+        converter2 = Converter(need_pad=True)
+
+        self.feature_scheme = {"text_tag": {
+                                    "extractor":  extractor1,
+                                    "converter": converter1
+                                },
+                                "ner_tag": {
+                                    "extractor":  extractor2,
+                                    "converter": converter2
+                                }
+                            }
+
+        class Model:
+            def __call__(self, tensor):
+                # Input shape: [Batch, length]
+                # Output shape: [Batch]
+                batch_size = len(tensor)
+                length = len(tensor[0])
+                return [[2 for _ in range(length)] for _ in range(batch_size)]
+        self.model = Model()
+
+    def convert(self, feature_collections):
+        example_collection = {}
+        for features_collection in feature_collections:
+            for tag, feature in features_collection.items():
+                if tag not in example_collection:
+                    example_collection[tag] = []
+                example_collection[tag].append(feature)
+
+        tensor_collection = {}
+        for tag, features in example_collection.items():
+            converter = self.feature_scheme[tag]["converter"]
+            tensor, mask = converter.convert(features)
+
+            if tag not in tensor_collection:
+                tensor_collection[tag] = {}
+
+            tensor_collection[tag]["tensor"] = tensor
+            tensor_collection[tag]["mask"] = mask
+        return tensor_collection
+
+    def yield_batch(self, pack):
+        for instance in pack.get(self.scope):
```

Comment:
It is possible that a scope is not provided, in which case we should use the whole pack.